### PR TITLE
Credential helpers

### PR DESCRIPTION
**Pull request by TheIndifferent**
Previous PR seems to be stalled, here is the same code with the `if` condition fix to invoke credentials helper if no credentials were found, plus merged with current master.

This functionality is tested on Docker-for-Mac and works fine. Unfortunately I am not able to test it on Windows, but it would be awesome if this was included in next release (as per "release early, release often") and maybe the Windows fix will arrive later when someone will have a chance to test/fix it?

See https://github.com/testcontainers/testcontainers-java/pull/647 for previous discussions.

### CODE REVIEW

**Comment by TheIndifferent** [comment]
Ok, didn't expect it to fail the build, sorry. Working on it.

**Comment by rnorth** [comment]
@TheIndifferent thanks - work isn't stalled per se, but I'm very busy 😄.

I'm very grateful for your input on this, and will close my original PR.

**Comment by TheIndifferent** [comment]
Ok, so quick update: the assumption about incorrect logic there is completely wrong. I am testing with `gcr.io` at the moment, it behaves little differently than expected, so I will try to add one more unit test and make everything work.

**Comment by TheIndifferent** [comment]
Please see the latest commit, had to reorganize the code a little bit, but now tested it with `gcloud docker --authorize-only` and with empty `auth` section but with helper present.

**Comment by TheIndifferent** [comment]
Cannot really understand why Travis build is failing, does not seem to be related. Could someone please take a look and either rebuild or point me to the actual error to fix?

**Comment by bsideup** [comment]
@TheIndifferent could be related to #731

**Comment by bsideup** [comment]
@TheIndifferent err, no, it's actually `javadoc` task (you can also run it locally)

**Comment by TheIndifferent** [comment]
@bsideup thank you for the pointer, fixed it, it passes now. Could you please take a look at current version of `RegistryAuthLocator` if it looks sane?

File: CHANGELOG.md [hunk 1]
```diff
@@ -22,7 +22,10 @@ All notable changes to this project will be documented in this file.
 
 ## [1.7.2] - 2018-04-30
 
+- Add support for private repositories using docker credential stores/helpers (fixes [\#567](https://github.com/testcontainers/testcontainers-java/issues/567))
```
Comment (rnorth) [inline CHANGELOG.md]: Please could you shift this up to the `UNRELEASED` section?

Also, it looks like this same line appears (more or less the same) three times - merge issue?

I think that for now we should draw a line in the sand and release this for Mac and Linux - could we say so in the changelog entry?

File: core/src/main/java/org/testcontainers/utility/RegistryAuthLocator.java [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+     *     <li>if no {@code credsStore} is found then the default configuration is returned.</li>
+     * </ol>
+     *
+     * @param dockerImageName image name to be looked up (potentially including a registry URL part)
+     * @return an AuthConfig that is applicable to this specific image OR the defaultAuthConfig that has been set for
+     * this {@link RegistryAuthLocator}.
+     */
+    public AuthConfig lookupAuthConfig(DockerImageName dockerImageName) {
+        log.debug("Looking up auth config for image: {}", dockerImageName);
+
```
Comment (rnorth) [inline core/src/main/java/org/testcontainers/utility/RegistryAuthLocator.java]: We should probably have a check here that we're not running on Windows and fall back to normal (current) behaviour.

**Comment by TheIndifferent** [comment]
@rnorth got it, I will address the comment this weekend, sorry for the delay.

**Comment by rnorth** [comment]
I have a moment now - I'll update this.

**Comment by TheIndifferent** [comment]
@rnorth thank you for picking this up and merging! Can't wait for next release to try it out for real :)

**Comment by rnorth** [comment]
Released in [1.8.1](https://github.com/testcontainers/testcontainers-java/releases/tag/1.8.1)! 🎉

Thanks for your work on this, @TheIndifferent!

**Pull request merged**